sdk/formrecognizer/ai-form-recognizer/test/utils/matrix.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Takes a jagged 2D array and a function and runs the function with every
+ * possible combination of elements of each of the arrays
+ *
+ * For strong type-checking, it is important that the `matrix` have a strong
+ * type, such as a `const` literal.
+ *
+ * @param values jagged 2D array specifying the arguments and their possible
```

Comment:
I am just curious what happens if higher dimensional arrays are passed to matrix()

Reply:
Anything that is nested deeper than 2 will become an argument itself.

```typescript
matrix([
  [["a", "b", "c"], ["x", "y", "z"]],
  [1, 2, 3]
] as const, (arr: string[], n: number) => {
  // ...
}); 
```